css handles for profile page

Dracenco opened the pull request:
#### What did you change? \*

Added custom css handles for profile section

#### Why? \*

Necessary for re-alining the content of the page as per customer design

#### How to test it? \*

Inspecting the page will reveal all the classes applied to "Profile" page

#### Related to / Depends on?

https://github.com/vtex-apps/my-account/pull/168

#### Types of changes \*

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Technical improvements
  <!--- * Required -->

salesfelipe commented on .idea/.gitignore:
```diff
@@ -0,0 +1,2 @@
+# Default ignored files
```
Plz remove it

salesfelipe commented on .idea/inspectionProfiles/Project_Default.xml:
```diff
@@ -0,0 +1,6 @@
+<component name="InspectionProjectProfileManager">
```
Plz remove it

salesfelipe commented on .idea/misc.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Plz remove it

salesfelipe commented on .idea/modules.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Plz remove it

salesfelipe commented on .idea/my-account.iml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Plz remove it

salesfelipe commented on .idea/vcs.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Plz remove it and also de yarn-erro.log

salesfelipe commented on react/components/Profile/NewsletterBox.tsx:
```diff
@@ -8,6 +8,7 @@ import ContentBox from '../shared/ContentBox'
 import GET_NEWSLETTER from '../../graphql/getNewsletterOpt.gql'
 import NEWSLETTER_MUTATION from '../../graphql/setOptInNewsletter.gql'
 import styles from '../../styles.css'
+import className from "../../styles/ContentBox.module.css";
```
Since this is the NewsletterBox I would change it to a different file
```suggestion
import className from "../../styles/NewsletterBox.css";
```

salesfelipe commented on react/components/Profile/PasswordBox.tsx:
```diff
@@ -9,6 +9,7 @@ import {
 import ContentBox from '../shared/ContentBox'
 import DataEntry from '../shared/DataEntry'
 import styles from '../../styles.css'
+import className from "../../styles/ContentBox.module.css";
```
Same thing about here...

salesfelipe replied:
```suggestion
import passwordBoxStyles from "../../styles/PasswordBox.css";
```

salesfelipe commented on react/components/Profile/ProfileBox.tsx:
```diff
@@ -5,6 +5,7 @@ import { ProfileRules, ProfileSummary } from 'vtex.profile-form'
 
 import ContentBox from '../shared/ContentBox'
 import DataEntry from '../shared/DataEntry'
+import className from '../../styles/ContentBox.module.css'
```
```suggestion
import styles from '../../styles/ContentBox.css'
```

salesfelipe commented on react/components/Profile/PasswordFormBox.tsx:
```diff
@@ -17,6 +17,7 @@ import ContentBox from '../shared/ContentBox'
 import RedefinePasswordForm from './RedefinePassword'
 import SendAccCodeButton from './SendAccCodeButton'
 import PasswordValidator from './PasswordValidator'
+import className from "../../styles/ContentBox.module.css";
```
ditto.

salesfelipe commented on react/components/Profile/ProfileBox.tsx:
```diff
@@ -43,38 +45,38 @@ const ProfileBox: FunctionComponent<Props> = ({
                 </div>
-                <div className="mb8 w-50-ns">
+                <div className={`mb8 w-50-ns ${className.lastNameSubContainer}`}>
                   <DataEntry label={lastName.label}>{lastName.value}</DataEntry>
                 </div>
               </div>
-              <div className="mb8">
+              <div className={`mb8 ${className.emailContainer}`}>
                 <DataEntry label={email.label}>{email.value}</DataEntry>
               </div>
-              <div className="flex-ns flex-wrap">
+              <div className={`flex-ns flex-wrap ${className.genderContainer}`}>
                 {document?.label && (
-                  <div className="mb8 flex-auto">
+                  <div className={`mb8 flex-auto ${className.documentsSubContainer}`}>
                     <DataEntry label={document.label}>
                       {document.value}
                     </DataEntry>
                   </div>
                 )}
-                <div className="mb8 w-50-ns">
+                <div className={`mb8 w-50-ns ${className.genderSubContainer}`}>
                   <DataEntry label={gender.label}>{gender.value}</DataEntry>
                 </div>
               </div>
-              <div className="flex-ns flex-wrap">
-                <div className="mb8 flex-auto">
+              <div className={`flex-ns flex-wrap ${className.phoneNumberContainer}`}>
+                <div className={`mb8 flex-auto ${className.dateOfBirthSubContainer}`}>
                   <DataEntry label={birthDate.label}>
                     {birthDate.value}
                   </DataEntry>
                 </div>
-                <div className="mb8 w-50-ns">
+                <div className={`mb8 w-50-ns ${className.phoneNumberSubContainer}`}>
```
Very nice ✨

salesfelipe commented on react/components/shared/ContentBox.tsx:
```diff
@@ -11,11 +13,11 @@ const ContentBox: FunctionComponent<Props> = ({
-    <div className={`pb5 pr5-ns ${flexClass} ${widthClass}`}>
-      <article className="ba bw1 b--muted-4 br2 flex flex-column justify-between">
-        <main className="ph7 pv6">{children}</main>
+    <div className={`pb5 pr5-ns test ${flexClass} ${widthClass}`}>
```
```suggestion
    <div className={`pb5 pr5-ns ${flexClass} ${widthClass}`}>
```

salesfelipe commented on react/components/shared/ContentBox.tsx:
```diff
@@ -1,5 +1,7 @@
 import React, { FunctionComponent, ReactElement } from 'react'
 import { Button } from 'vtex.styleguide'
+import className from '../../styles/ContentBox.module.css'
```
```suggestion
import styles from '../../styles/ContentBox.css'
```

salesfelipe commented on react/components/shared/DataEntry.tsx:
```diff
@@ -2,10 +2,10 @@ import React, { FunctionComponent } from 'react'
 const DataEntry: FunctionComponent<Props> = ({ label, children }) => {
   return (
-    <div>
+    <>
```
I have a better proposal, why don't you add prop, cssHandle and pass the css handle to here like:

```suggestion
    <div className={cssHandle}>
```

salesfelipe commented on react/styles/ContentBox.module.css:
```diff
@@ -0,0 +1,43 @@
+.boxContainer {}
```
Change file to `ContentBox.css`

Dracenco commented:
@salesfelipe I have added a log to the changelog file and also re-named the css file but there are 2 issues with your other comments that I could not fix.
1. I do not have .idea folder at all , including everything that you mentioned inside it
2.     Changing empty tag to a div is burrying the entry data fields inside each section that has any information in it. I can show you on a call why it is actually making things more difficult to have the div there. Besides until now it didn't have any class at all so I did not see any use for it.
<>
<div className={cssHandle}>

please let me know if there is anything else to address

salesfelipe commented:
> * I do not have .idea folder at all , including everything that you mentioned inside it
> *

This is probably created by default by your IDE, that's why you can't see it. We recommend using VSCode: https://code.visualstudio.com/

We have a bunch of integrations for it.

Dracenco commented:
@salesfelipe removed the content from .idea files

salesfelipe commented on react/components/Profile/NewsletterBox.tsx:
```diff
@@ -8,6 +8,7 @@ import ContentBox from '../shared/ContentBox'
 import GET_NEWSLETTER from '../../graphql/getNewsletterOpt.gql'
 import NEWSLETTER_MUTATION from '../../graphql/setOptInNewsletter.gql'
 import styles from '../../styles.css'
+import className from "../../styles/ContentBox.css";
```
```suggestion
import className from '../../styles/ContentBox.css'
```

salesfelipe commented on react/components/Profile/ProfileFormBox.tsx:
```diff
@@ -9,6 +9,7 @@ import { Button } from 'vtex.styleguide'
 import ContentBox from '../shared/ContentBox'
 import UpdateProfile from '../../graphql/updateProfile.gql'
 import { withSettings, Settings } from '../shared/withSettings'
+import className from "../../styles/ContentBox.css";
```
```suggestion
import className from '../../styles/ContentBox.css'
```

This is probably a formatting problem with your prettier: 

https://www.jetbrains.com/help/idea/prettier.html

https://www.jetbrains.com/help/idea/eslint.html

salesfelipe commented:
I still the IDE files on the PR, also can you remove the `node/yarn-error.log`

![image](https://user-images.githubusercontent.com/3926634/85458442-7eb8d700-b577-11ea-9356-f8866a8c8b99.png)

salesfelipe commented:
It seems some of the modified code don't adhere to VTEX's code style guide. Can you run `yarn lint --fix` and `yarn format` to fix these issues? Also, can you remove the .idea directory from the git history?

Dracenco commented:
@salesfelipe  I have fixed the lint errors and also removed the .idea directory from the latest commit. however the only way I found to remove a directory from the history requires me to still have the directory. Do you have any other solution you could suggest for me to get rid of the folder in git history ? sorry if it is a silly question

salesfelipe commented:
> @salesfelipe I have fixed the lint errors and also removed the .idea directory from the latest commit. however the only way I found to remove a directory from the history requires me to still have the directory. Do you have any other solution you could suggest for me to get rid of the folder in git history ? sorry if it is a silly question

Hi, no worries, this is not silly. You can do this to remove: 

https://stackoverflow.com/questions/2047465/how-can-i-delete-a-file-from-a-git-repository

And then add an entry (`.idea`) to the `.gitignore` file, so you will be able to still have this folder locally, but will not go to our git repo.

Dracenco commented:
.idea is a part of .gitignore. I am assuming it wasn't when I was using phpstorm. now using VS it is there. as for the rm --cached file_name all the paths are returning empty
<img width="1792" alt="Screenshot 2020-06-24 at 16 21 09" src="https://user-images.githubusercontent.com/16210306/85564309-d36f5700-b636-11ea-9531-dd091f1ddf87.png">

salesfelipe commented:
> .idea is a part of .gitignore. I am assuming it wasn't when I was using phpstorm. now using VS it is there. as for the rm --cached file_name all the paths are returning empty
> <img alt="Screenshot 2020-06-24 at 16 21 09" width="1792" src="https://user-images.githubusercontent.com/16210306/85564309-d36f5700-b636-11ea-9531-dd091f1ddf87.png">

Try `.idea/`

Dracenco commented:
farm3@192-168-0-102 my-account % git rm --cached .idea/
fatal: pathspec '.idea/' did not match any files
farm3@192-168-0-102 my-account % git rm .idea/         
fatal: pathspec '.idea/' did not match any files
farm3@192-168-0-102 my-account % git rm idea/          
fatal: pathspec 'idea/' did not match any files
farm3@192-168-0-102 my-account % git rm --cached idea/ 
fatal: pathspec 'idea/' did not match any files

salesfelipe commented on CHANGELOG.md:
```diff
@@ -7,12 +7,15 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+- Added custom css Handles to profile section 
```
```suggestion
### Added

- Custom css Handles to profile section 
```

salesfelipe replied:
Last changes, I promise 🙏

salesfelipe commented on react/components/Profile/PasswordBox.tsx:
```diff
@@ -9,6 +9,7 @@ import {
 import ContentBox from '../shared/ContentBox'
 import DataEntry from '../shared/DataEntry'
 import styles from '../../styles.css'
+import className from "../../styles/ContentBox.css";
```
```suggestion
import className from '../../styles/ContentBox.css'
```

salesfelipe commented on react/components/Profile/PasswordFormBox.tsx:
```diff
@@ -17,6 +17,7 @@ import ContentBox from '../shared/ContentBox'
 import RedefinePasswordForm from './RedefinePassword'
 import SendAccCodeButton from './SendAccCodeButton'
 import PasswordValidator from './PasswordValidator'
+import className from "../../styles/ContentBox.css"
```
```suggestion
import className from '../../styles/ContentBox.css'
```

Dracenco commented:
@salesfelipe  I have update the changelog as requested . is there anything I missed ? 
<img width="1792" alt="Screenshot 2020-06-24 at 18 46 09" src="https://user-images.githubusercontent.com/16210306/85588069-04598700-b64b-11ea-8f56-0782f2fb541f.png">
